### PR TITLE
docs(readme): add information about terraform [CI SKIP]

### PR DESCRIPTION
**Pull request by MrSimonEmms**
## Ticket Number

AS-

## Description of change

Add docs about Terraform

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [x] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -197,6 +197,53 @@ Or:
     git add .
     npm run commit
 
+# Terraform
+
+[Terraform](https://www.terraform.io/) is used to provision the infrastructure in the Azure. The state is stored in
```
Comment (a6software) [inline README.md:202]: > in the Azure

did you accidentally a word?

File: README.md [hunk 2]
```diff
@@ -197,6 +197,53 @@ Or:
...
+> only use that version. If you use a more recent version, this will update the state file (stored in Azure) and mean that
+> the CI/CD pipeline will need upgrading to that version too.
+
+There are two pieces of infrastructure that are configured that do different things:
+
+## Common
+
+[Source](/infrastructure/common)
+
+This is things common to all environments, such as the container registry. There is only a [single workspace](https://www.terraform.io/docs/language/state/workspaces.html)
```
Comment (a6software) [inline README.md:215]: > This is things

These are things ?

File: README.md [hunk 3]
```diff
@@ -197,6 +197,53 @@ Or:
...
+here (`default`). It is unlikely that this will need to be changed. If development is to be attempted in here, please
+understand all the implications of making a change - as there is only a single workspace, this may affect all users and
+deployed instances.
+
+## Environments
+
+[Source](/infrastructure/environments)
+
+This is the individual environments, currently `dev`, `preprod` and `prod` (plus any environments for individual developers).
+Each of these are allocated to a different workspace (eg, `dev`). All these
```
Comment (a6software) [inline README.md:225]: > All these

...

File: README.md [hunk 4]
```diff
@@ -197,6 +197,53 @@ Or:
...
+
+## Terraform Commands
+
+It is a good idea to use [direnv](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/Using-direnv-for-local-development)
+for managing local environment variables.
+
+In the PINS active directory, you will need to create a [Service Principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals).
```
Comment (frank-dodion) [inline README.md:232]: Does this mean we are missing a Service Principal currently?

File: README.md [hunk 5]
```diff
@@ -197,6 +197,53 @@ Or:
...
+The one used by CI/CD is called "Terraform" and additional secrets can be generated to grant access, if you have the credentials
```
Comment (frank-dodion) [inline README.md:233]: Can we link to a doc in Confluence confirming who has these permissions or what policy applies?

File: README.md (hunk 4 above)
Comment (MrSimonEmms) [reply README.md:232]: No, we have one. The line after that details it

File: README.md [hunk 6]
```diff
@@ -197,6 +197,53 @@ Or:
     git add .
     npm run commit
 
+# Terraform
+
+[Terraform](https://www.terraform.io/) is used to provision the infrastructure in the Azure. The state is stored in 
+Azure Blob Storage.
+
+> **Important** - this application is built using [Terraform v0.14.0](https://releases.hashicorp.com/terraform/0.14.0/). Please
+> only use that version. If you use a more recent version, this will update the state file (stored in Azure) and mean that
```
[the remark this answers is not in the document]
Comment (MrSimonEmms) [reply README.md:206]: Oooh, this is new. Good catch

**Review by Russ-Clarke — APPROVED** [review]

**Review by Fenisay — APPROVED** [review]

**Review by a6software — APPROVED** [review]

**Review by frank-dodion — APPROVED** [review]

**Pull request merged**